cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -92,7 +92,16 @@ type Etcd struct {
 	DataDir   string
 	ExtraArgs map[string]string
 	// Image specifies which container image to use for running etcd. If empty, automatically populated by kubeadm using the image repository and default etcd version
-	Image string
+	Image      string
+	SelfHosted SelfHostedEtcd
+}
+
+type SelfHostedEtcd struct {
```

Comment:
let's do better than the rest of the file and add comments everywhere :)

Reply:
+100 to that!